### PR TITLE
add ssl logging support

### PR DESCRIPTION
**Pull request by aiwantaozi**
add ssl support for elasticsearch, splunk, kafka.

Related: types: https://github.com/rancher/types/pull/467/files
Related issues: 
https://github.com/rancher/rancher/issues/13065
https://github.com/rancher/rancher/issues/13304
https://github.com/rancher/rancher/issues/13526
https://github.com/rancher/rancher/issues/13560
https://github.com/rancher/rancher/issues/13760

### CODE REVIEW

File: pkg/controllers/user/logging/utils/secret.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+var (
+	initSecretData = map[string][]byte{
+		loggingconfig.CaFileName:     []byte(""),
+		loggingconfig.ClientCertName: []byte(""),
+		loggingconfig.ClientKeyName:  []byte(""),
+	}
+)
+
+func UpdateSSLAuthentication(prefix, target string, esConfig *v3.ElasticsearchConfig, spConfig *v3.SplunkConfig, kfConfig *v3.KafkaConfig, secrets rv1.SecretInterface) error {
+	var certificate, clientCert, clientKey string
```
Comment (alena1108) [inline pkg/controllers/user/logging/utils/secret.go]: I feel like target is a bit redundant here. The config type can be determined by either of esConfig/spConfig/kfConfig not being nil

File: pkg/controllers/user/logging/utils/secret.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	return nil
+}
+
+func updateSecret(name, prefix, certificate, clientCert, clientKey string, secrets rv1.SecretInterface) error {
+	existSecret, err := secrets.Get(name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			data := map[string][]byte{
+				prefix + "_" + loggingconfig.CaFileName:     []byte(certificate),
```
Comment (alena1108) [inline pkg/controllers/user/logging/utils/secret.go]: lines 77-79 are repeated at 92-94, can we perhaps extract it to its own method?

File: pkg/controllers/user/logging/utils/secret.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	newSecret := existSecret.DeepCopy()
+	newSecret.Data[prefix+"_"+loggingconfig.CaFileName] = []byte(certificate)
+	newSecret.Data[prefix+"_"+loggingconfig.ClientCertName] = []byte(clientCert)
+	newSecret.Data[prefix+"_"+loggingconfig.ClientKeyName] = []byte(clientKey)
+
+	if reflect.DeepEqual(existSecret.Data, newSecret.Data) {
+		return nil
+	}
+
+	if _, err = secrets.Update(newSecret); err != nil {
```
Comment (alena1108) [inline pkg/controllers/user/logging/utils/secret.go]: is the secret ever being removed? I guess there are no implications of it being left around, but still

Comment (aiwantaozi) [reply pkg/controllers/user/logging/utils/secret.go]: add reset when related crd is removed, the secret is deployed in cattle-logging namespace, when there is no target config, the namespace will be deleted.

**Pull request merged**